project-base/storefront/components/Basic/GoogleMap/GoogleMap.tsx

```diff
@@ -0,0 +1,142 @@
+            {children}
+        </div>
+    </div>
+);
+
+export const GoogleMap: FC<GoogleMapProps> = ({
+    lat,
+    lng,
+    zoom: zoomProps,
+    markers,
+    activeMarkerHandler,
+    isDetail,
+    closeMarkers,
+}) => {
+    const { publicRuntimeConfig } = getConfig();
+    const { mapSetting } = useDomainConfig();
+    const defaultLat = lat === null || lat === undefined ? mapSetting.latitude : parseFloat(lat);
+    const defaultLng = lng === null || lng === undefined ? mapSetting.longitude : parseFloat(lng);
+    const defaultZoom = zoomProps === null || zoomProps === undefined ? mapSetting.zoom : zoomProps;
+    const [activeMarkerId, setActiveMarkerId] = useState('');
+    const [zoom, setZoom] = useState(defaultZoom);
+    const [bounds, setBounds] = useState<number[]>([]);
+
+    const points = useMemo(
+        () =>
+            markers?.map((marker) => ({
+                type: 'Feature',
+                properties: {
+                    cluster: false,
+                    markerId: marker.identifier ?? marker.uuid,
+                },
+                geometry: {
+                    type: 'Point',
+                    coordinates: [
+                        marker.locationLongitude !== null ? parseFloat(marker.locationLongitude) : null,
+                        marker.locationLatitude !== null ? parseFloat(marker.locationLatitude) : null,
+                    ],
+                },
+            })) ?? [],
+        [markers],
+    );
+
+    const { clusters } = useSupercluster({
+        points,
+        zoom,
+        bounds,
+        options: { radius: 75, maxZoom: 20 },
+    });
+
+    const markerClickHandler = (id: string) => {
+        if (isDetail) {
+            return;
+        }
+        const newActiveMarkerId = activeMarkerId === id ? '' : id;
+        setActiveMarkerId(newActiveMarkerId);
+        activeMarkerHandler?.(newActiveMarkerId);
+    };
+
+    useEffect(() => {
+        if (!closeMarkers) {
+            return;
+        }
+        setActiveMarkerId('');
+    }, [closeMarkers]);
+
+    return (
+        <div className="h-full w-full" data-testid={TEST_IDENTIFIER}>
+            <GoogleMapReact
+                yesIWantToUseGoogleMapApiInternals
+                bootstrapURLKeys={{ key: publicRuntimeConfig.googleMapApiKey }}
+                defaultCenter={{ lat: defaultLat, lng: defaultLng }}
+                defaultZoom={defaultZoom}
+                options={{
+                    disableDoubleClickZoom: true,
+                    fullscreenControl: false,
+                    zoomControlOptions: { position: 1 },
+                }}
+                onChange={({ zoom, bounds }) => {
+                    setZoom(zoom);
+                    setBounds([bounds.nw.lng, bounds.se.lat, bounds.se.lng, bounds.nw.lat]);
+                }}
+            >
+                {clusters.map((cluster) => {
```

Comment:
The things inside this `map` could be a separate component.

Reply:
Done.